gc.c

```diff
@@ -225,7 +225,46 @@ int sexp_valid_object_p (sexp ctx, sexp x) {
 #define sexp_gc_pass_ctx(x)
 #endif
 
-void sexp_mark_one (sexp_gc_pass_ctx(sexp ctx) sexp* types, sexp x) {
+struct sexp_mark_stack_ptr_t {
+  sexp *start, *end;
+  struct sexp_mark_stack_ptr_t *next; /* TODO: remove for allocations on stack */
+};
+struct sexp_mark_stack_ptr_t *sexp_mark_stack_ptr = NULL;
+
+/* TODO: make resizable */
```

Comment:
This can go in features.h as a compile-time option, I don't think we need to worry about runtime.

Reply:
Ok, will move the stack count there.